### ISSUE TITLE
Reconnect repository should insert Deploy Key to Github

### ISSUE THREAD

**Issue by equivalent** [aws-amplify/amplify-console#1837]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-console/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-console/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-console/blob/master/CONTRIBUTING.md).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### App Id

arn:aws:amplify:eu-west-2:694647969796:apps/dg1s64s7fmyba

### Region

eu-west-2

### Amplify Console feature

Build settings

### Describe the bug

Given I  use Github with AWS Amplify 
When I select repository
Then "Deployment Key" is inserted to this repository on Github
:+1:  so far so good

Given I remove this "Deployment Key" from Github repository (e.g. due to https://github.com/aws-amplify/amplify-console/issues/1830) 
When I Reconnect Repsitory (in AWS Amplify > General settings)
Then deployment key is not reinserted  :-1: 

### Expected behavior

Given I remove  "Deployment Key" from Github repository
When I Reconnect Repsitory (in AWS Amplify > General settings)
Then deployment key should be reinserted

### Reproduction steps

1. configure AWS Amplify to use Github Repo
2. go to Github repository settings > Deployment Keys > remove the key inserted by AWS Amplify
3. go to AWS Amplify > general settings > reconnect repsitory
4. check  Github repository settings > Deployment Keys  == see no new deployment key

### Additional information

this is  related to bug  https://github.com/aws-amplify/amplify-console/issues/1830#

and https://github.com/aws-amplify/amplify-console/issues/157#issuecomment-834362708  and https://github.com/aws-amplify/amplify-console/issues/1815

**Comment by JKalm** [comment]
Hello Tomas,

I think I was able to find a possible cause. When you click Reconnect Repository the message below is shown
![image](https://user-images.githubusercontent.com/39815749/118335967-9c5f9980-b4c5-11eb-9727-fc8ff905df82.png)
Make sure to follow instructions and delete existing webhooks, if you proceed without this - request silently fails.

Thank you for feedback, I will file a bug to show explicit message on UI when webhooks are not removed during reconnect. Please let us know if that was the cause.

**Comment by equivalent** [comment]
Hi @JKalm, thank you for looking into this

> When you click Reconnect Repository the message below is shown

In my case when I try to reconnect - reconnect happens successfully - no error appears, Github say a-ok, AWS Amplify say a-ok. I think the issue you are describing is unrelated to what I'm describing 

the core of the issue as I've described in this issue screenshots the deployment keys are no longer there => AWS amplify app is no longer able to clone the repo. There is no problem with reconnecting the App to repo. It's just that the reconnect will not add new deployment keys.

BTW a developer is able to connect as many AWS Amplify Apps to the same repo . Don't know why webhooks would   be an issue. But his is not important.

> I will file a bug to show explicit message on UI when webhooks are not removed during reconnect

that is  a reasonable solution but again this is not the issue. In the case of the deployment keys they should be readded if they are missing once `re-connect` is clicked


> Please let us know if that was the cause.

It was not,

as from my side I wont be able to spend more time reproducing this issue for you (already spent 2 weeks , raised 3 issues related to this. I'm pretty sure we pinpoint exact moment when the error happens) We done this research just to see if AWS Amplify is stable enough product to be used with our company. Now that we know what was the problem we will carry on with other work.

 In our company we simply decided we will be careful not to de-authorize the application ( https://github.com/aws-amplify/amplify-console/issues/1830) and not remove deployment keys or mess around with users while we are doing our tests.

But as described in this issue  or issue https://github.com/aws-amplify/amplify-console/issues/1830 or  all the comments from various other developers in issue https://github.com/aws-amplify/amplify-console/issues/157#issuecomment-834362708 AWS Amplify have issues working with Github this may discourage other Companies from using it. So my only advice is to look deeper into what I'm actually describing in this issue for the good name of AWS Amplify.

**Comment by ferdingler** [comment]
Hi @equivalent, I understand your frustration, let me give you some context on what's going on here. The webhook and the deploy keys are indeed related (as part of the process of connecting a repository, we create both), so if you delete the deploy keys, you'll need to delete the webhook as well in order to re-connect the repository again to your Amplify app. If you don't delete the Webhook, our backend will not be able to re-connect the repository. 

While this is a rare edge case, we acknowledge this is not the best user experience so we have filed a bug into our backlog to make this process of re-connecting a repository better. Thank you for bringing this to our attention!

**Comment by equivalent** [comment]
@ferdingler @JKalm thank you both  for the answer. I will try your solution next time this error happens. My apologies if my previous comment sounded harsh. I'm grateful for the work done on AWS Amplify.

**Comment by JohnHershberg** [comment]
I'm also having this issue. Reconnect doesn't do anything. I have no web hook and no deploy key in my github repo. No idea how to add them back in. Can those be added back in manually?

**Comment by DSSLab-code** [comment]
Also have this error. I have already dig for a day.

**Comment by LucasAndrad** [comment]
I have this same error, @equivalent and @JohnHershberg described exactly what is the problem.

**Comment by LucasAndrad** [comment]
@ferdingler any updates on this one?

**Comment by LucasAndrad** [comment]
Not sure if this is related, but I started to have this problem after the user who connected the Github repo with Amplify was removed from our Github organization. Since then, I'm having the same problem described by @JohnHershberg.

**Comment by equivalent** [comment]
@LucasAndrad my opinion is that this will not be fixed in near future (or not so near future)

What we done in our team is that we created fresh Github user, we connected all SPAs from scratch and we  told every developer in the team **not to ever ever ever disconnect/reconnect/remove/touch anything with those SPAs on Github** configuration or in AWS Amplify (we have just one person responsible to do changes in repo setting) 

Now the team can do changes on Amplify itself like adding Redirects or ENV variables or Auto branches - just not to try to change Repo connections

Yep it's a stupid and fragile way around it (one Junior developer hire close to screw-up or production) but so far so good.

But I agree with you that this is an existing issue and it should be fixed

**Comment by equivalent** [comment]
One more note if AWS Amplify ever implement this https://github.com/aws-amplify/amplify-console/issues/1815  (SSH keys) then I'm 80% sure this will solve this issue as well

**Comment by LucasAndrad** [comment]
Yes @equivalent I agree, it's very frustrating to pay for a service and sunddely start to face issues like this one with no feedback/support from the team. I already reported many issues on amplify projects and I never got any feedback with less than 3 months (unreal right ??) I decided it doesn't worth any more, Amplify it's just for very very simple MVPs, I don't recommend to use it with real projects, not even the small ones. I moved all my stuff out of it, even for deployment I started using S3.

**Comment by oli-xentraffic** [comment]
Facing the same issue still, its pretty blocking.
